tests/test_desktop_control.py

```
@@ -0,0 +1,163 @@
+    broker = DesktopCommandBroker()
+    q = await broker.subscribe("user-1")
+    n = DesktopCommandBroker._MAX_QUEUE + 5
+    for i in range(n):
+        await broker.emit("user-1", DesktopCommand(kind="open-app", payload={"app": f"app-{i}"}))
+    assert q.qsize() == DesktopCommandBroker._MAX_QUEUE
+    # The newest command must still be in the queue (oldest were dropped).
+    apps = []
+    while not q.empty():
+        apps.append(q.get_nowait().payload["app"])
+    assert apps[-1] == f"app-{n - 1}"
+    assert f"app-0" not in apps
```

Ruff (0.15.15)

[error] f-string without any placeholders

Remove extraneous `f` prefix

(F541)